test/hotspot/jtreg/compiler/c2/TestMinMaxSubword.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2022, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.Utils;
+import java.util.Random;
+
+/*
+ * @test
+ * @bug 8294816
+ * @summary Test Math.min/max vectorization miscompilation for integer subwords
+ * @library /test/lib /
+ * @requires vm.compiler2.enabled
+ * @run driver compiler.c2.TestMinMaxSubword
+ */
+
+public class TestMinMaxSubword {
+    private static final int LENGTH = 256;
+    private static final Random RANDOM = Utils.getRandomInstance();
+    private static int val = 65536;
+    private static short[] sa;
+    private static short[] sb;
+    private static byte[] ba;
+    private static byte[] bb;
+
+    static {
+        sa = new short[LENGTH];
+        sb = new short[LENGTH];
+        ba = new byte[LENGTH];
+        bb = new byte[LENGTH];
+        for(int i = 0; i < LENGTH; i++) {
+            sa[i] = (short) (RANDOM.nextInt(999) + 1);
+            ba[i] = (byte) (RANDOM.nextInt(99) + 1);
+        }
+    }
+
+    // Ensure vector max/min instructions are not generated for integer subword types
+    // as Java APIs for Math.min/max do not support integer subword types and superword
+    // should not generate vectorized Min/Max nodes for them.
+    @Test
+    @IR(failOn = {IRNode.Min_V})
```

Comment:
The test currently fails compilation because the IRNodes were renamed by https://github.com/openjdk/jdk/pull/10695 to ` MIN_V`.

Reply:
Thanks for letting me know. I have updated the code accordingly. Please review..